docs/language/informal/toplevel-inference.md

```diff
@@ -0,0 +1,293 @@
+`void`.
+
+
+## Static variable and field inference
+
+
+A “candidate” for inference is any of the following:
+* A top level variable definition
+* A static class level variable definition
+* An instance variable, getter or setter definition with the type annotation
+  omitted.
+
+
+The *inference dependencies of a candidate* which is inferred via override
+inference are the instance fields, setters, and getters that the candidate
+overrides.
+
+
+The *inference dependencies of a candidate* which is inferred via initializer
+inference are the *inference dependencies of the initializer expression* used to
+infer the type of the candidate.
+
+
+The *inference dependencies of an expression*  are as defined below
+as part of initializer inference.
+
+
+Informally, the inference dependencies of a candidate are all of the program
+elements that contribute to inference for the candidate, and hence which must
+have their types inferred before the candidate can have its type inferred.  Note
+that by design, a top level variable or static class variable never depends on
+an instance field, setter, or getter.  This means that top level variable and
+static class variable inference can be done without reference to the results of
+field inference.
+
+
+For each candidate, inference for all of its dependencies must be performed
+before inference for the candidate is performed.  If this cannot be done
+(because of a cycle) it is a static error.  Note that it is always possible for
+the programmer to break a cycle and hence eliminate the error by adding a type
+annotation.
+
+
+### Candidate inference
+
+#### Top level variables
+
+The inferred type of a top level variable is the type inferred from its
+initializer.
+
+#### Static class variables
+
+The inferred type of a static class variable is the type inferred from its
+initializer.
+
+#### Instance field, getter, and setter inference
+
+
+The inferred type of a getter, setter, or field is as follows.  Note that I say
+that a setter overrides a getter if there is a getter of the same name in some
+superclass or interface, and similarly for setters overriding getters, fields,
+etc.
+
+
+A getter, setter or field which overrides/implements only a getter is inferred
+to have the type taken from the overridden getter result type.
+
+
+A getter, setter or field which overrides/implements only a setter is inferred
+to have the type taken from the overridden setter parameter.
+
+
+A getter which overrides/implements both a setter and a getter is inferred to
+have the type taken from the overridden getter result type.
+
+
+A setter which overrides/implements both a setter and a getter is inferred to
+have the type taken from the overridden setter parameter type.
+
+
+A field which overrides/implements both a setter and a getter is inferred to
+have the type taken from the overridden setter parameter type if this type is
+the same as the return type of the overridden getter (if the types don’t match
```

Comment:
Does "match" mean "the same"?
If so, it would be to write so explicitly to prevent any doubts.